enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,322 @@
+  )
+}
+
+resource "azurerm_resource_group" "main" {
+  tags     = local.tags
+}
+```
+
+The tag of the form `kubernetes.io_cluster.<cluster_id>:owned` added by openshift, where
+cluster_id is a string generated by concatenating user inputted cluster name and a random 
+string will be limited to a maximum length of 27 characters by trimming long cluster name 
+to 21 characters.
+
+#### Caveats
+1. Updating or removing resource tags added by Openshift using an external interface,
+   may or may not be reconciled by the operator managing the resource.
+2. Updating tags of individual resources is not supported and any tag present in 
+   `.status.platformStatus.azure.resourceTags` of `infrastructure.config.openshift.io/v1` resource 
+   will result in adding tags to all Openshift managed Azure resources. 
+
+### Risks and Mitigations
+
+### Drawbacks
+- User-defined tags cannot be updated on an Azure resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in scope.
+  User-defined tags can be updated on the following Azure resources.
+    1. Virtual machine resources created for master and worker nodes.
+    2. Storage account
+    3. NetworkEdge
+    4. Internal Registry
+
+- OpenShift is bound to have the common limitation for all Azure resources created
+  by it and constraints other resources with the least matching limit as below
+    1. Tag names cannot have `microsoft`, `azure`, `windows` prefixes which are
+       reserved for Azure use.
+    2. An Azure storage account has a limit of 128 characters for the tag name.
+    3. An Azure DNS zone or Traffic or Front Door resource tag name cannot have spaces,
+       special/unicode characters or start with a number.
+    4. An Azure Automation or Content Delivery Network or DNS resource can have a
+       maximum of 15 tags.
+
+- Administrator will have to manually perform below tags pertaining actions
+    1. removing the undesired tags from the required resources.
+    2. update tag values of the required resources.
+    3. update tags of the resources which are not managed by an operator.
+    4. update tags of the resources for which update logic is not supported by an operator.
+
+## Design Details
+
+### Open Questions
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+- Feature available for end-to-end usage.
+- Complete end user documentation.
+- UTs and e2e tests are present.
+- Gather feedback from the users.
+
+#### Tech Preview -> GA
+N/A. This feature is for Tech Preview, until decided for GA.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+On upgrade:
+- Cluster operators that update the tags of Azure resources created for cluster 
+  should refer the new fields and take action. Any new resource created post-upgrade
+  and the operators managing the resource will add the user defined tags to the 
```

Comment:
Some copy edits:
```suggestion
- Cluster operators that update the tags of Azure resources created for the cluster 
  should refer to the new fields and take action. For any new resource created post-upgrade,
  the operator managing the resource will add the user-defined tags to the 
```
I was initially confused by the phrasing "operators that update the tags of Azure resources created for cluster"; does that refer to operators that update tags on existing resources when `userTags` is updated?
```suggestion
- Cluster operators that update tags on existing resources when `userTags` is updated
  should refer to the new fields and take action. For any new resource created post-upgrade,
  the operator managing the resource must add the user-defined tags to the 
```

Reply:
I meant to convey, those operators which manage an azure resource and add tags should refer to the new fields. I will rephrase it to mean adding tags to azure resources.